Add service broker space scope information to service wall list

richard-cox opened the pull request:
- fixes #4458
- ux needs a good review

mogul commented:
Showing the broker name in the column mirrors what happens on the command-line, and it represents the data about brokers accurately, but it's not very useful to users. When looking at the Marketplace, it's more important to know 
1. If this service **isn't** available in all spaces, what spaces **is** it available in?
2. If this service is listed multiple times, what's the difference between them?

Also, because space-scoped brokers have to have unique names even if they're in different spaces, and maybe people are testing out their own version of the broker, you have to add a bunch of noise to the broker name to avoid collisions. For example, mine looks like this:
```
% cf service-brokers
Getting service brokers as bret.mogilefsky@gsa.gov...

name                                                   url
ssb-development-<username>   https://solid-turkey.app.cloud.gov
ssb-staging-<username>       https://solid-turkey.app.cloud.gov
```

The username is actually the name of the deployer account, which is a GUID! So this ends up pretty ugly when you see the brokername in the output of `cf marketplace`.

What would be more useful: Have a column for "Scope", where the value is either "All spaces", or a list of the spaces where that service offering is available. Consolidate offerings that are registered more than once.

The user already selects the target space from among those allowed during the configuration workflow... It'll just be a smaller list for space-scoped stuff registered more than once.

richard-cox commented:
@mogul I've made some tweaks following a review and your comments

1) Scope has been split out and has it's own row (card view) or column (table view)
2) Added broker scope info to the service instances list (both views). For the table view the `service` cell is pretty bloated now, however I felt it's nice to have the extra info there too (we show the space it's in regardless)
3) Added broker scope info to the application services list (card view). 

For 2 & 3 we may rejig these to look nicer given changes required to support v3 of the CF API (both content and requests have changed for all service related items).

> What would be more useful: Have a column for "Scope", where the value is either "All spaces", or a list of the spaces where that service offering is available. Consolidate offerings that are registered more than once.

I like this idea, but it would be tricky with v2 of the CF API. As it stands we dispatch a request per service broker to fetch broker info for the services on screen. In order to group by scope we'd need to fetch service brokers for all services which would create a scaling issue. It might be easier in v3 but would require some investigation. It may be best to detail this request in a separate issue to be dealt with at the time we update to v3.
